lighthouse-core/gather/gatherers/http-redirect.js

```diff
@@ -36,39 +36,20 @@ class HTTPRedirect extends Gatherer {
   }
 
   afterPass(options) {
-    // Reset the options.
-    options.url = this._preRedirectURL;
-
-    // Allow override for faster testing.
-    const timeout = options._testTimeout || 10000;
-
-    const securityPromise = options.driver.getSecurityState()
-      .then(state => {
-        return {
-          value: state.schemeIsCryptographic
-        };
+    let isRedirect = false;
+
+    if(options.driver._httpsArr.length !== 0) {
+      const httpsExpectURL = options.url.replace(/^http/, 'https');
+      options.driver._httpsArr.forEach( (httpsUrl) => {
+        if(httpsUrl.url === httpsExpectURL) {
```

Comment:
I'd say we don't want to enforce that they redirect to the exact entered URL just with a scheme change, for example when testing `http://theverge.com` it ultimately sends you to `https://www.theverge.com/` which seems fine. @brendankenny did you have a different scenario in mind that you wanted exact equality?